strawberryfields/backends/tfbackend/backend.py

```diff
@@ -128,14 +128,14 @@ def reset(self, pure=True, **kwargs):
                 If False, then the circuit is reset to its initial state, but ops that
                 have already been declared are still accessible.
         """
-        hard = kwargs.get('hard', True)
+        hard = kwargs.pop('hard', True)
```

Comment:
Just out of curiosity, what is `hard`? Does it refer to a hard reset?

Reply:
yes

Reply:
otherwise, it will make a separate circuit simulation, but in the same underlying graph (i.e., the old circuit objects are still in tensorflow graph)